src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+
+    $output_callback('out', 'Installing Composer production dependencies');
+    $this->localMachineHelper->execute(['composer', 'install', '--no-dev', '--no-interaction', '--optimize-autoloader'], $output_callback, $artifact_dir, $this->output->isVerbose());
+  }
+
+  protected function sanitize(Closure $output_callback, string $artifact_dir):void {
+    $output_callback('out', 'Finding Drupal core text files');
+    $sanitizeFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->name('*.txt')
+      ->notName('LICENSE.txt')
+      ->in("$artifact_dir/docroot/core");
+
+    $output_callback('out', 'Finding VCS directories');
+    $vcsFinder = $this->localMachineHelper->getFinder()
+      ->ignoreDotFiles(FALSE)
+      ->ignoreVCS(FALSE)
+      ->directories()
+      ->in(["$artifact_dir/docroot",
+        "$artifact_dir/vendor",
+      ])
+      ->name('.git');
+    $drush_dir = "$artifact_dir/drush";
+    if (file_exists($drush_dir)) {
+      $vcsFinder->in($drush_dir);
+    }
+    if ($vcsFinder->hasResults()) {
+      $sanitizeFinder->append($vcsFinder);
+    }
+
+    $output_callback('out', 'Finding INSTALL database text files');
+    $dbInstallFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->in([$artifact_dir])
+      ->name('/INSTALL\.[a-z]+\.(md|txt)$/');
+    if ($dbInstallFinder->hasResults()) {
+      $sanitizeFinder->append($dbInstallFinder);
+    }
+
+    $output_callback('out', 'Finding other common text files');
+    $filenames = [
+      'AUTHORS',
+      'CHANGELOG',
+      'CONDUCT',
+      'CONTRIBUTING',
+      'INSTALL',
+      'MAINTAINERS',
+      'PATCHES',
+      'TESTING',
+      'UPDATE',
+    ];
+    $textFileFinder = $this->localMachineHelper->getFinder()
+      ->files()
+      ->in(["$artifact_dir/docroot"])
+      ->name('/(' . implode('|', $filenames) . ')\.(md|txt)$/');
+    if ($textFileFinder->hasResults()) {
+      $sanitizeFinder->append($textFileFinder);
+    }
+
+    $output_callback('out', "Removing sanitized files from build");
+    $this->localMachineHelper->getFilesystem()->remove($sanitizeFinder);
+  }
+
+  protected function commit(Closure $output_callback, string $artifact_dir, string $commit_hash):void {
+    $output_callback('out', 'Adding and committing changed files');
+    $this->localMachineHelper->execute(['git', 'add', '-A'], $output_callback, $artifact_dir, $this->output->isVerbose());
+    foreach (array_merge($this->vendorDirs($artifact_dir), $this->scaffoldFiles($artifact_dir)) as $file) {
+      // This will fatally error if the file doesn't exist. Suppress error output.
+      $this->logger->debug("Forcibly adding $file");
+      $this->localMachineHelper->execute(['git', 'add', '-f', $file], NULL, $artifact_dir, FALSE);
+    }
+    $this->localMachineHelper->execute(['git', 'commit', '-m', "Automated commit by Acquia CLI (source commit: $commit_hash)"], $output_callback, $artifact_dir, $this->output->isVerbose());
```

Comment:
It seems like we are adding a new System dependency for git. should we check for it before we start the process?

Reply:
`LocalMachineHelper::commandExists()` isn't very reliable in its current state. I think checking for a common dependency like Git is more likely to do harm than good. I think we have tickets to both improve `LocalMachineHelper::commandExists()` and to create a Git wrapper, those would be the place to add a check for Git.